src/data-sources/plugins/ApplicationInsights/Query.ts

```diff
@@ -186,6 +196,30 @@ export default class ApplicationInsightsQuery extends DataSourcePlugin<IQueryPar
     return typeof query === 'function' ? query(dependencies) : query;
   }
 
+  private compileQueryWithFilters(query: any, dependencies: any, isForked: boolean, queryKey: string, filters: IFilterParams[]): string {
```

Comment:
Yes, I pass isForked to apply the right logic. Here are the parts that change given two un-forked filter datasources 'filterChannels' and 'filterIntents':

In 'ai' datasource:
```js
			selectedChannels: "filterChannels:channels-selected",
			selectedIntents: "filterIntents:intents-selected"
```
In filters (sample for 'filterChannels' DS):
```js
dependencies: {
			selectedValues: "filterChannels:channels-selected",
			values: "filterChannels:channels-filters"
		},
		actions: {
			onChange: "filterChannels:updateSelectedValues"
		},
```

However one point of difference to note when using an un-forked datasource is that the **calculated** state needs mapped using `var { filterChannels } = state;` or `var { filterIntents } = state;`.